src/pages/Home/Home.page.jsx

```diff
@@ -1,38 +1,58 @@
-import React, { useRef } from 'react';
-import { Link, useHistory } from 'react-router-dom';
+import React, { useEffect, useState } from 'react';
+import { WelcomeTitle, ContentDiv, TitleDiv, ContDiv, ContDivFst, PrevBtn, NextBtn} from './Home.page.styled'
+
+import HomeContent from './Content/HomeContent';
 
-import { useAuth } from '../../providers/Auth';
 import './Home.styles.css';
 
-function HomePage() {
-  const history = useHistory();
-  const sectionRef = useRef(null);
-  const { authenticated, logout } = useAuth();
+function HomePage({ txtSearch }) {
+  const useYoutubeApi = (url) => {
+    const [data, setData] = useState();
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+      const fetchData = async () => {
+        const response = await fetch(url);
+        const data = await response.json();
+        setIsLoading(false);
+        setData(data);
+      };
+      fetchData();
+    }, [url]);
+
+    return [data, isLoading];
+  };
+
+  const term = txtSearch;
+
+  const [data, isLoading] = useYoutubeApi(
+    `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=${process.env.REACT_APP_MAX_RESULTS_SEARCH}&key=${process.env.REACT_APP_API_KEY}&q=${term}&type=video`
```

Comment:
I think the URL can be moved to the custom hook and just pass a query parameter, the rest can live in the hook.